File: Algorithms/Warmup/diagonalDifference.py
#!/bin/python3
# https://www.hackerrank.com/challenges/diagonal-difference
import math
import os
import random
import re
import sys

#
# Complete the 'diagonalDifference' function below.
#
# The function is expected to return an INTEGER.
# The function accepts 2D_INTEGER_ARRAY arr as parameter.
#

def diagonalDifference(arr):
    return abs(primaryDiagonalSum(arr)-secondaryDiagonalSum(arr))

def primaryDiagonalSum(arr):
    sum = 0
    for i in range(0, len(arr)):
        sum += arr[i][i]   
    return sum

def secondaryDiagonalSum(arr):
    sum = 0
    for i in range(0, len(arr)):
        for j in range(0, len(arr)):
            if (i + j == len(arr)-1):
                sum += arr[i][j]
    return sum

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    n = int(input().strip())

    arr = []

    for _ in range(n):
        arr.append(list(map(int, input().rstrip().split())))

    result = diagonalDifference(arr)

    fptr.write(str(result) + '\n')

    fptr.close()

File: Algorithms/Warmup/diagonalDifference.js
/*
 * Complete the 'diagonalDifference' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts 2D_INTEGER_ARRAY arr as parameter.
 */

function diagonalDifference(arr) {
    let n = arr.length;
    let mainDiagonal = 0, antiDiagonal = 0;
    for (let i = 0; i < n; i++) {
        for (let j = 0; j < n; j++) {
            // Main diagonal
            if (i === j) {
                mainDiagonal += arr[i][j];
            }
            
            // Anti diagonal condition
            if ((i + j) === (n-1)) {
                antiDiagonal += arr[i][j];
            }
        }
    }

    let res = mainDiagonal - antiDiagonal;
    return (res > 0) ? res : (res * -1);
}

let arr = [[1, 2, 3], [4, 5, 6], [9, 8, 9]];
console.log(diagonalDifference(arr));